packages/lib/server/getLuckyUser.ts

```diff
@@ -130,18 +119,113 @@ function getUsersWithHighestPriority<T extends Pick<User, "id" | "email"> & { pr
   );
 }
 
+async function getUsersBasedOnWeights<
+  T extends PartialUser & {
+    weight?: number | null;
+    weightAdjustment?: number | null;
+  }
+>({
+  availableUsers,
+  bookingsOfAvailableUsers,
+  allRRHosts,
+  eventType,
+}: GetLuckyUserParams<T> & { bookingsOfAvailableUsers: PartialBooking[] }) {
+  //get all bookings of all other RR hosts that are not available
+  const availableUserIds = new Set(availableUsers.map((user) => user.id));
+  const notAvailableHosts = allRRHosts.filter((host) => !availableUserIds.has(host.user.id));
+
+  const bookingsOfNotAvailableUsers = await BookingRepository.getAllBookingsForRoundRobin({
+    eventTypeId: eventType.id,
+    users: notAvailableHosts.map((host) => {
```

Comment:
Instead of iterating through `allRRHosts` and then iterating through the subset of that array, can we use a single reduce here?